securedrop/journalist_app/utils.py

```diff
@@ -134,21 +142,26 @@ def validate_hotp_secret(user: Journalist, otp_secret: str) -> bool:
     :param otp_secret: the new HOTP secret
     :return: True if it validates, False if it does not
     """
+    strip_whitespace = otp_secret.replace(' ', '')
+    secret_length = len(strip_whitespace)
+
+    if secret_length != HOTP_SECRET_LENGTH:
+        flash(ngettext(
+                'HOTP secrets are 40 characters long - you have entered {num}.',
+                'HOTP secrets are 40 characters long - you have entered {num}.',
```

Comment:
I mean, there's a good argument for it, if we envisaged that changing. Considering the limit is baked into Yubikeys (which are all we recommend for HOTP) and gives a good 160 bits of key length (which is the upper limit recommended by the relevant RFC and probably overkill from a cryptographic perspective), I don't think it will change, and as such it's probably not worth changing the string and triggering work for translators.

Reply:
Okay, I'm with you (a) that `HOTP_SECRET_LENGTH` is highly unlikely to change and (b) that we want to avoid needlessly adding or complicating translation work.  Something like—

```python
        flash(ngettext(
                f'HOTP secrets are {HOTP_SECRET_LENGTH} characters long - you have entered {{num}}.',
                f'HOTP secrets are {HOTP_SECRET_LENGTH} characters long - you have entered {{num}}.',
                secret_length
            ).format(num=secret_length), "error")
```

—would, I believe, defer to `HOTP_SECRET_LENGTH` (rather than hard-coding its value of `40`) *and* still satisfy these desiderata.  But I'm happy to downgrade this suggestion from "reality check" to "nit not worth picking".  :-)